### PR TITLE
LiB flow: set the site title, tagline and logo url for every patterns

### PR DESCRIPTION
**Pull request by arthur791004**
#### Proposed Changes

* Update the content of LiB patterns
* Set the site title, tagline and logo URL to each pattern

https://user-images.githubusercontent.com/13596067/187895167-cdf80758-8fc6-4b5f-8666-45e021c8ec70.mov

#### Testing Instructions

* Apply D87019-code to your sandbox
* Go through the LiB flow via http://calypso.localhost:3000/setup?flow=link-in-bio
* Check the title, tagline, and logo on each pattern are correct

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #

### CODE REVIEW

**Review by alshakero — CHANGES_REQUESTED** [review]
This is definitely a step forward, but I think we shouldn't hard-code more stuff on the frontend. Because it's extra bytes and a maintenance burden on all of us.

[In this issue](https://github.com/Automattic/wp-calypso/issues/67195), we attempt to make the API the source of truth, but we're blocked by the patterns not being accessible for unauthenticated users. 

I think the way should this:

1. Allow access to patterns for everyone. 
2. Move the default titles and tag lines logic to the backend (the client side isn't making any decisions here).
3. Remove the hardcoded patterns from the client side.

**Comment by arthur791004** [comment]
Yes, I agree with you. But I think it should not be a blocker for this one. We need to make these patterns work on LiB flow, and we could create a follow-up PR to make data come from the endpoint. What do you think?

BTW, even if we can get patterns from the endpoint, I have no idea to set the default site title, tagline, and logo for each pattern as those data do not belong to the pattern itself. Do you have an idea for this part?

File: client/landing/stepper/hooks/use-setup-onboarding-site.ts [hunk 1]
```diff
@@ -119,9 +107,8 @@ export function useSetupOnboardingSite( options: SetupOnboardingSiteOptions ) {
 		if ( ( ignoreUrl || shouldSetupOnboardingSite() ) && site && flow ) {
 			return Promise.all( [
 				postSiteSettings( site, state ),
-				postSiteLogo( state ).then( ( logoUploadResult ) =>
-					setPattern( site, flow, logoUploadResult )
-				),
+				postSiteLogo( state ).then( () => Promise.resolve() ),
```
Comment (alshakero) [inline client/landing/stepper/hooks/use-setup-onboarding-site.ts]:
```suggestion
				postSiteLogo( state ),
```

**Review by alshakero — APPROVED** [review]
OK this works as intended. We can follow up as you said.

Just noting that the pattern is applied but the site looks off when you visit it. Probably unrelated at all.

File: client/landing/stepper/hooks/use-setup-onboarding-site.ts (hunk 1 above)
Comment (arthur791004) [reply client/landing/stepper/hooks/use-setup-onboarding-site.ts]: The problem here is the return value of `postSiteLogo( state )` is not a Promise and doesn't match the argument type of `Promise.all` 😅

**Comment by simison** [comment]
> BTW, even if we can get patterns from the endpoint, I have no idea to set the default site title, tagline, and logo for each pattern as those data do not belong to the pattern itself. Do you have an idea for this part?

Would it make sense to record these at the post for the pattern somehow?

File: client/landing/stepper/hooks/use-setup-onboarding-site.ts (hunk 1 above)
Comment (alshakero) [reply client/landing/stepper/hooks/use-setup-onboarding-site.ts]: I just double-checked and it does return a promise.

**Comment by alshakero** [comment]
@simison I think Arthur means he doesn't know how to set these at the **preview** API backend. We'd need a map somewhere in the backend that has the pattern slug and the information it needs to placehold site title and tagline. 

Something like this:

```php
'lynx/link-in-bio-background-image' => array('siteTitle' => 'Emily Jennings', 'siteTagline' => 'Leadership coaching for leaders'),
```

**Comment by simison** [comment]
@alshakero  Yep, I was suggesting you'd use the pattern's post meta as "database" to store that info, and fetch+structure it for the preview API from there?

That still poses some issues with translations, and isn't necessarily the "simple" solution we'd prefer, nor I don't know how that would be for designer workflows.

**Comment by alshakero** [comment]
Oh that's actually a good idea!

**Comment by arthur791004** [comment]
Oh, that's interesting! Let me investigate how to do it later

File: client/landing/stepper/hooks/use-setup-onboarding-site.ts (hunk 1 above)
Comment (arthur791004) [reply client/landing/stepper/hooks/use-setup-onboarding-site.ts]: After checking again, I found the issue is the value is non-void. So, I add the type definition to each promise as below
```diff
- Promise.all
+ Promise.all< void, SetSiteLogoResponse | void, void, void, void >
```

File: packages/pattern-picker/src/item.tsx [hunk 2]
```diff
@@ -11,23 +12,39 @@ type Props = {
...
 };
 
+const getPatternMetaValue = ( patternMeta: string, regexp: RegExp ) => {
+	const [ , value ] = patternMeta.split( regexp );
+	if ( value ) {
+		return value.replace( /%2C/g, ',' );
+	}
+
+	return null;
+};
```
Comment (alshakero) [inline packages/pattern-picker/src/item.tsx]: I would make this function `parsePatternMetaData` and parse the whole data into a key value object. Having regexes outside this function is kind of leaky.

**Review by alshakero — COMMENTED** [review]
Great work!

File: packages/pattern-picker/src/item.tsx (hunk 2 above)
Comment (arthur791004) [reply packages/pattern-picker/src/item.tsx]: Okay, that makes sense to me!

Comment (arthur791004) [reply packages/pattern-picker/src/item.tsx]: I've made a new function `parsePatternMetaData` and moved this one to the internal function. What do you think?

File: packages/pattern-picker/src/item.tsx [hunk 3]
```diff
@@ -11,6 +12,70 @@ type Props = {
...
+		if ( siteTitle ) {
+			result.siteTitle = siteTitle;
+		}
+
+		if ( siteTagline ) {
+			result.siteTagline = siteTagline;
+		}
+
+		if ( siteLogoUrl ) {
+			result.siteLogoUrl = siteLogoUrl;
```
Comment (alshakero) [inline packages/pattern-picker/src/item.tsx]: If you keep this in kebab case, you can pass them directly to the URLSearchParams like shown below.

File: packages/pattern-picker/src/item.tsx [hunk 4]
```diff
@@ -11,6 +12,70 @@ type Props = {
...
+		params.set( 'site_title', siteTitle );
+	}
+
+	if ( siteTagline ) {
+		params.set( 'site_tagline', siteTagline );
+	}
+
+	if ( siteLogoUrl ) {
+		params.set( 'site_logo_url', siteLogoUrl );
+	}
```
Comment (alshakero) [inline packages/pattern-picker/src/item.tsx]:
```suggestion
		pattern_id: [ pattern.ID, pattern.site_id ].join( '-' ),
		...parsePatternMetaData( pattern.pattern_meta ),
	} );
```

File: packages/pattern-picker/src/use-query-patterns.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import { useQuery, UseQueryResult } from 'react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+import type { Pattern } from './types';
+
+const useQueryPatterns = ( siteSlug: string ): UseQueryResult< Pattern[] > => {
+	const locale = useLocale();
+	const params = new URLSearchParams( {
+		site: siteSlug,
+		tags: 'pattern',
+		http_envelope: '1',
```
Comment (alshakero) [inline packages/pattern-picker/src/use-query-patterns.ts:11]: I think this is automatically added.

**Review by alshakero — APPROVED** [review]
I can't thank you enough. Really amazing work.

File: packages/pattern-picker/src/use-query-patterns.ts (hunk 5 above)
Comment (arthur791004) [reply packages/pattern-picker/src/use-query-patterns.ts:11]: No, the default one is `_envelope` and it's not compatible with this API namespace version 😂

**Pull request merged**